### PR TITLE
Fixes hair and facial hair not blocking emissives as they should

### PR DESCRIPTION
**Pull request by GoldenAlpharex**
## About The Pull Request
Basically, everything that composes the sprites of mobs blocks emissive.

Everything, except hair, for some reason, which leads to silly stuff like this being possible:
image
The bottom portion of the hair is blocking the emissives, because it's actually the head that's underneath that's blocking said emissives, whereas the top, isn't.

This PR aims to fix that, and it works just as expected.

## Why It's Good For The Game
Having parts of your body be completely illuminated when they shouldn't be makes them look really stupid.

## Changelog

:cl: GoldenAlpharex
fix: Fixed hair not blocking emissives, which ended up creating some rather silly visual glitches at times, in darker areas.
/:cl:

### CODE REVIEW

**Comment by Serxule** [comment]
i have no idea what that image is showing

**Comment by GoldenAlpharex** [comment]
> i have no idea what that image is showing

It's small on purpose, because I don't want people reeing in the comment because I couldn't be bothered to re-open my test-server on a different branch to take a picture, so it's from a downstream.

**Comment by magatsuchi** [comment]
image

File: code/modules/surgery/bodyparts/head.dm [hunk 1]
```diff
@@ -247,6 +247,7 @@
 				if(sprite_accessory)
 					//Create the overlay
 					facial_overlay = mutable_appearance(sprite_accessory.icon, sprite_accessory.icon_state, -HAIR_LAYER)
+					facial_overlay.overlays += emissive_blocker(facial_overlay.icon, facial_overlay.icon_state, alpha = facial_overlay.alpha)
```
Comment (Kapu1178) [inline code/modules/surgery/bodyparts/head.dm:250]: IIRC hair overlays actually get their alpha set later in a proc chain, so use the `hair_alpha` var. I think.

File: code/modules/surgery/bodyparts/head.dm [hunk 2]
```diff
@@ -259,6 +260,7 @@
 				sprite_accessory = GLOB.hairstyles_list[hair_style]
 				if(sprite_accessory)
 					hair_overlay = mutable_appearance(sprite_accessory.icon, sprite_accessory.icon_state, -HAIR_LAYER)
+					hair_overlay.overlays += emissive_blocker(hair_overlay.icon, hair_overlay.icon_state, alpha = hair_overlay.alpha)
```
Comment (Kapu1178) [inline code/modules/surgery/bodyparts/head.dm:263]: Same here

File: code/modules/surgery/bodyparts/head.dm (hunk 1 above)
Comment (GoldenAlpharex) [inline code/modules/surgery/bodyparts/head.dm:250]:
```suggestion
					facial_overlay.overlays += emissive_blocker(facial_overlay.icon, facial_overlay.icon_state, alpha = hair_alpha)
```

File: code/modules/surgery/bodyparts/head.dm (hunk 2 above)
Comment (GoldenAlpharex) [inline code/modules/surgery/bodyparts/head.dm:263]:
```suggestion
					hair_overlay.overlays += emissive_blocker(hair_overlay.icon, hair_overlay.icon_state, alpha = hair_alpha)
```

File: code/modules/surgery/bodyparts/head.dm (hunk 1 above)
Comment (GoldenAlpharex) [reply code/modules/surgery/bodyparts/head.dm:250]: I wonder if there's actually a facial_hair_alpha, now that I think about it

Comment (Kapu1178) [reply code/modules/surgery/bodyparts/head.dm:250]: I didn't make one because I didn't forsee it to even be possible for them to have seperate alphas.

Comment (GoldenAlpharex) [reply code/modules/surgery/bodyparts/head.dm:250]: Okay, good, this means I can just commit this without fearing any issue, good.

**Pull request merged**